provider/server/server_suite_test.go

```diff
@@ -38,38 +47,93 @@ func TestServer(t *testing.T) {
 	SetDefaultConsistentlyDuration(consistentlyDuration)
 
 	RegisterFailHandler(Fail)
-	RunSpecs(t, "Server Suite")
+	RunSpecs(t, "GRPC Server Suite", Label("integration"))
 }
 
 var _ = BeforeSuite(func() {
-	logf.SetLogger(zap.New(zap.WriteTo(GinkgoWriter), zap.UseDevMode(true)))
-
-	tmpDir := GinkgoT().TempDir()
-	machineStore, err := host.NewStore[*api.Machine](host.Options[*api.Machine]{
-		Dir: tmpDir,
-		NewFunc: func() *api.Machine {
-			return &api.Machine{}
-		},
-		CreateStrategy: utils.MachineStrategy,
-	})
-	Expect(err).NotTo(HaveOccurred())
+	logf.SetLogger(GinkgoLogr)
 
-	machineClasses, err := mcr.NewMachineClassRegistry([]iri.MachineClass{
+	machineClasses := []iriv1alpha1.MachineClass{
 		{
-			//TODO
 			Name: machineClassx3xlarge,
-			Capabilities: &iri.MachineClassCapabilities{
+			Capabilities: &iriv1alpha1.MachineClassCapabilities{
 				CpuMillis:   4000,
 				MemoryBytes: 8589934592,
 			},
 		},
-	})
+	}
+	machineClassData, err := json.Marshal(machineClasses)
+	Expect(err).NotTo(HaveOccurred())
+
+	machineClassesFile, err := os.CreateTemp(GinkgoT().TempDir(), "machineclasses")
+	Expect(err).NotTo(HaveOccurred())
+	defer func() {
+		_ = machineClassesFile.Close()
+	}()
+	Expect(os.WriteFile(machineClassesFile.Name(), machineClassData, 0666)).To(Succeed())
+
+	By("starting the app")
+
+	fs := pflag.NewFlagSet("test", pflag.ContinueOnError)
+	apinetPlugin := networkinterfaceplugin.NewDefaultOptions()
+	apinetPlugin.PluginName = "apinet"
+	apinetPlugin.AddFlags(fs)
+	Expect(fs.Set("apinet-node-name", "test-node")).To(Succeed())
+
+	tempDir := GinkgoT().TempDir()
+	Expect(os.Chmod(tempDir, 0777)).Should(Succeed())
```

Comment:
why 0777? it looks little dangerous and amateur.

Reply:
In the actual codebase, where the user's home directory serves as the root directory, special permissions are not required. However, during testing, we create a temporary directory which typically gets created at root's /tmp, requiring broader permissions for successful execution. I've optimized the tests to operate with the minimal required permissions of 0703.

Reply:
but still why everyone has permission for write and execute, it is technically more dangerous part of permission.

Reply:
Addressed, removed others permissions.

Reply:
how is it removed, when you returned worse version with 0777? Do you know how permissions on linux work?

Reply:
sorry, it was fixed. github opened to me wrong version.